src/gui/editors/PianoRoll.cpp

```diff
@@ -2595,6 +2612,41 @@ void PianoRoll::dragNotes( int x, int y, bool alt, bool shift, bool ctrl )
 			else
 			{
 				// moving note
+				if (m_gridMode == gridSnap && firstIteration)
+				{	
+					firstIteration = false;
+					Note* copy(m_draggedNote);
+					// guantize first note
+					int start_pos_ticks = copy->oldPos().getTicks() + unquantized_off_ticks;
+					int end_pos_ticks = start_pos_ticks + copy->length().getTicks();
+					start_pos_ticks = qMax(0, start_pos_ticks);
+					end_pos_ticks = qMax(0, end_pos_ticks);
+										
+					copy->setPos(TimePos(start_pos_ticks));
+					copy->quantizePos(quantization());
+					auto tmp_pos=copy->pos(); // store start position
+					
+					int gap_start = qAbs(start_pos_ticks - copy->pos().getTicks());
+
+					copy->setPos(TimePos(end_pos_ticks));
+					copy->quantizePos(quantization());
+
+					int gap_end = qAbs(end_pos_ticks - copy->pos().getTicks());
+					
+					if (gap_start < gap_end)
+					{
+						copy->setPos(tmp_pos); // restore start position
+						off_ticks += copy->pos().getTicks() - (m_draggedNote->oldPos().getTicks() + off_ticks);
+					}
+					else 
```

Comment:
Trailing whitespaces:
```suggestion
					else
```